### DISCUSSION TITLE
CJS/ESM Issue with @mui as nested dependency

### DISCUSSION THREAD

**Discussion by MikaelColliander** [vikejs/vike#1590] [Discussion]
Hi, any help in resolving my issue would be deeply appreciated.

With the plugin `vite-plugin-cjs-interop` I am able to use @mui/material within my Vike application code, directly. As suggested here: https://github.com/brillout/vike-with-mui/blob/main/vite.config.ts

However, I use packages that in turn depend on @mui packages which yields the CJS/ESM Issue error when running `npm run dev`.

Here is a reproduction repo:
https://github.com/MikaelColliander/vike-test/blob/main/layouts/LayoutDefault.tsx

Here is the relevant output: 

`node_modules/@sj-ab/component-library.ui.theme-provider/ThemeProvider.tsx:2
import { ThemeProvider as MuiThemeProvider } from "@mui/material/styles";
         ^
SyntaxError: The requested module '@mui/material/styles' does not provide an export named 'ThemeProvider'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at ModuleJob.run (node:internal/modules/esm/module_job:190:5)
┌────────────────────────────────────────────────────────────────────────────────┐
│ The error seems to be a CJS/ESM issue, see https://vike.dev/broken-npm-package │
└────────────────────────────────────────────────────────────────────────────────┘`

**Comment by MikaelColliander** [comment]
When adding to vite.config

`ssr: {
  noExternal: true,
}`

There is another error:

`ReferenceError: exports is not defined
    at /Users/mikaelcolliander/Development/my-app/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/server.node.js:12:1
    at instantiateModule (file:///Users/mikaelcolliander/Development/my-app/node_modules/.pnpm/vite@5.2.7/node_modules/vite/dist/node/chunks/dep-C-KAszbv.js:55006:15)`

**Comment by brillout** [comment]
Can you make it a minmal reproduction?

**Reply by MikaelColliander** [reply]
Thanks for taking time with this. I was thinking  this might serve as a minimal reproduction, will it suffice?
https://github.com/MikaelColliander/vike-test/tree/main

**Reply by brillout** [reply]
Minimal pacage.json#dependencies
